### ISSUE TITLE
double "/" cause http server to fail

### ISSUE THREAD

**Issue by bol-van** [gerbera/gerbera#323]
curl http://192.168.2.7:49152/description.xml

`<URLBase>http://192.168.2.7:49152/</URLBase>`

Then SAMSUNG TV queries

http://192.168.2.7:49152//cm.xml

 get 400 Bad request
and all futher queries come with double /

**Comment by bol-van** [comment]
its not gerbera issue, but libupnp

this allows samsung tv to browse folder :
```

		if (membuffer_append_str(&url_str, "http://") != 0 ||
		    membuffer_append_str(&url_str, ip_str) != 0 ||
/*		    membuffer_append_str(&url_str, "/") != 0 || */
		    membuffer_append_str(&root_path, "/") != 0) {
			goto error_handler;
		}

```

**Comment by v00d00** [comment]
Hi @bol-van, thanks for reporting.

Can you report this upstream with https://github.com/mrjimenez/pupnp?

Thanks!

**Comment by TimelessNL** [comment]
Any progress on this?
This is probably why my 2012 TV in [#352](https://github.com/gerbera/gerbera/issues/352#issuecomment-612445906) are not working I suppose.
MediaTomb seems to had a [workarounds](https://sourceforge.net/p/mediatomb/patches/37/) in place as well. To remove excessive slashes:
```
+
+        //remove excessive leading slashes (fix for Samsung Smart TV 2012)
+        while( ( ( begin_path + 1 ) < max ) && ( in[begin_path] == '/' ) && ( in[begin_path + 1] == '/') ) {
+            begin_path++;
+        }
+
```

**Comment by pamapa** [comment]
is this still an issue with the latest master version of gerbera?

**Comment by TimelessNL** [comment]
I've tested Gerbera v1.4.0 and libupnp v1.8.6 and that needed the fix. I could test master is there any reason to believe that it has been sorted?

**Comment by pamapa** [comment]
I have landed a fix (https://github.com/pupnp/pupnp/commit/dc2c54bd710ba9f824c85b8a2a8bd66496767378) to address this issue within libupnp, should be part of the next release (v1.12.2).

**Comment by pamapa** [comment]
fixed in pupnp when using a recent release: 1.12.1 or higher
